contest/scoreboard.py

```diff
@@ -0,0 +1,139 @@
+'''
+    The MIT License (MIT)
+    Copyright (c) 2014 NTHUOJ team
+    Permission is hereby granted, free of charge, to any person obtaining a copy
+    of this software and associated documentation files (the "Software"), to deal
+    in the Software without restriction, including without limitation the rights
+    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+    copies of the Software, and to permit persons to whom the Software is
+    furnished to do so, subject to the following conditions:
+    The above copyright notice and this permission notice shall be included in all
+    copies or substantial portions of the Software.
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+    SOFTWARE.
+    '''
+import sys
+from operator import methodcaller
+class Scoreboard:
+    def __init__(self,start_time):
+        self.users = []
+        self.problems = []
+        self.start_time = start_time
+
+    def add_user(self,user):
+        self.users.append(user)
+
+    def add_problem(self,scoreboard_problem):
+        self.problems.append(scoreboard_problem)
+
+    def get_problem(self,id):
+        for scoreboard_problem in self.problems:
+            if (scoreboard_problem.id == id):
+                return scoreboard_problem
+
+    #sort by solved descending. if same sort by penalty
+    def sort_users_by_penalty(self):
+        self.users = sorted(self.users, key=methodcaller('penalty',self.start_time))
+        self.users = sorted(self.users, key=methodcaller('solved'), reverse=True)
+
+    def sort_users_by_solved_testcases(self):
+        self.users = sorted(self.users, key=methodcaller('testcases_solved'), reverse=True)
+
+#for scoreboard
+class Scoreboard_Problem:
```

Comment:
Please use ScoreboardProblem for Python class naming rule.
The naming is clearly enough, you may remove comment in previous line.